fbsat-core/src/main/kotlin/ru/ifmo/fbsat/core/task/single/complete/Cegis.kt

```diff
@@ -129,6 +127,38 @@ fun Inferrer.performCegis(smvDir: File, loopNumber: Int): Automaton? {
             )
             return null
         }
+
+        val maxLabel = if (Globals.NEGATIVE_TREE_OPTIMIZATIONS == NegativeTreeOptimizations.OPT1) {
+            val prevMaxLabel = negativeScenarioTree.nodes.maxOfOrNull { it.label }
+            if (prevMaxLabel != null) {
+                val visitedNodes = mutableSetOf<NegativeScenarioTree.Node>()
+                val visitedFullScenarios = negativeScenarioTree.scenarios.filter { scenario ->
+                    val path = automaton.eval(scenario).toList()
```

Comment:
Are you sure that it is impossible to use lazy data structure (`Sequence`) returned by `eval`? One day this might shoot you in the foot, if you will continue allocating lists for extremely large scenarios.